src/components/Item/VisualizationItem/Visualization/__tests__/Visualization.spec.js

```diff
@@ -0,0 +1,134 @@
+import React from 'react'
+import { render } from '@testing-library/react'
+import { Provider } from 'react-redux'
+import configureMockStore from 'redux-mock-store'
+
+import Visualization from '../Visualization'
+
+jest.mock('@dhis2/data-visualizer-plugin', () => 'VisualizationPlugin')
+jest.mock('../MapPlugin', () => 'MapPlugin')
+jest.mock('../DefaultPlugin', () => 'DefaultPlugin')
+
+const mockStore = configureMockStore()
+
+test('renders a MapPlugin when activeType is MAP', () => {
+    const store = {
+        visualizations: { rainbowVis: { rows: [], columns: [], filters: [] } },
+    }
+    const { container } = render(
+        <Provider store={mockStore(store)}>
+            <Visualization
+                item={{
+                    id: 'rainbow',
+                    type: 'MAP',
+                    chart: { id: 'rainbowVis' },
+                }}
+                activeType="MAP"
+                itemFilters={{}}
+                availableHeight={500}
+            />
+        </Provider>
+    )
+    expect(container).toMatchSnapshot()
+})
+
+test('renders a VisPlugin when activeType is CHART', () => {
+    const store = {
+        visualizations: { rainbowVis: { rows: [], columns: [], filters: [] } },
+    }
+    const { container } = render(
+        <Provider store={mockStore(store)}>
+            <Visualization
+                item={{
+                    id: 'rainbow',
+                    type: 'REPORT_TABLE',
```

Comment:
Why doesn't this and the next test have the same `type` as `activeType`? 
`type: 'REPORT_TABLE'` and `activeType="CHART"` 🤔

Reply:
This is simulating the "View as" feature of dashboards. E.g., viewing a chart as a table, or a table as a chart. I've changed the test name so that it's hopefully more clear.